### ISSUE TITLE
Please don't store plugin config files directly under the user's home directory

### ISSUE THREAD

**Issue by SpotlightKid** [sjoerdvankreel/firefly-synth#7]
Firefly stores a file `plugin.xml` under `/home/<username>/Sjoerd van Kreel/Firefly Synth/<some ID>/1.0/`. This apparently gets created even when the plugin is only loaded via scanning for pluigins by the host

Please respect Linux conventions and the [XDG Base Directory](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) spec and store configuration files under `XDG_CONFIG_HOME` (default: `~/.config`), so the user's home directory doesn't get cluttered with random directories.

**Comment by ofsjoerdk** [comment]
I'm using JUCE to handle all the per-user settings storage so this might take a bit of tweaking.
Just to confirm:

- Is this the problem? https://forum.juce.com/t/userapplicationdatadirectory-why-doesnt-this-use-xdg-data-home/11458
- And does something like this fix it? https://github.com/surge-synthesizer/stochas/issues/85

**Comment by SpotlightKid** [comment]
* Yes.
* Yes, most probably.

**Comment by sjoerdvankreel** [comment]
Allright, thanks. I'll look into it.

**Comment by sjoerdvankreel** [comment]
Fixed in 1.7.3, can you confirm?

**Comment by SpotlightKid** [comment]
Yes, I can confirm that the VST3 plugins now save their config files under `/home/<user>/.config/Sjoerd van Kreel/Firefly Synth (FX)`.

**Comment by SpotlightKid** [comment]
However, when I try to actually load the Firefly Synth or Firefly Synth FX VST3 plugin in Carla, it crashes with a segmentation fault. Previous version worked fine.

Both load ok in Ardour (8.6), though.

And the CLAP version work in Carla too.

I compiled the plugins on my system (Manjaro) from latest git using `cd scripts; sh build_linux.sh Release`.

**Comment by SpotlightKid** [comment]
Just found the links to the pre-compiled binaries in the README (was looking for them on the git tag page), and installed the synth VST3. Same crash when loading the plugin in Carla.

**Comment by sjoerdvankreel** [comment]
Thanks for reporting. TBH i only test linux using reaper and bitwig. Will try out using Carla.

**Comment by sjoerdvankreel** [comment]
Notes to self:

* VST3 looks OK on carla windows
* carla ubuntu 22 refuses to scan, apparently i need to update https://linuxmusicians.com/viewtopic.php?t=24514

**Comment by sjoerdvankreel** [comment]
@SpotlightKid ok so I've managed to install carla 2.4.3 (apparently 2.4.2 has a known bug wrt vst3 scanning) inside ubuntu 22 on virtualbox. Now, I don't know the first thing about carla, but it "seems to work". At least i can do add plugin->firefly, have the carla controls for firefly show up, and also have the firefly gui itself show up. So the plugin definately loads.

I really don't hope this is a distro-specific thing. I suspect something in the XDG support is the culprit, but it might be anything honestly.

My test setup:
* firefly 1.7.3 VST3
* ubuntu 22.04.2 LTS under virtualbox
* carla 2.4.3

I'm ok with installing a manjaro VM but i'd need some specifics:
* whats your carla version ? (also if its not 2.4.3 could you try with that?)
* whats the latest version of firefly that worked, and first that broke ?
* what manjaro version ?

![image](https://github.com/sjoerdvankreel/firefly-synth/assets/2647549/cff08ea1-7410-4f11-98e6-b04c75c990e6)

**Comment by SpotlightKid** [comment]
Ok, I checked, and I actually had a Carla version compiled from the Git HEAD installed, which identified itself as 2.6.0-alpha1 in the About dialog.

The current version of Carla is 2.5.8 and that's what's available on Arch Linux/Manjaro. So I installed that via the Manjaro package management and, behold, that version loaded Firefly Synth VST3 and Firefly Synth FX VST3 just fine. That version of Carla does not have support for CLAP yet, though.

Interestingly, though, I was also using the 2.6.0-alpha1 version when I tested Firefly before I opened this ticket and that worked. I must have compiled that version of Firefly from Git, which was the current HEAD around the time I opened the ticket. Sorry, I can't tell exactly anymore which commit that was.

To sum it up: problem exists only in latest Carla from Git and VST3 format and it broke sometime between the opening of this ticket and yesterday.

Feel free to ignore this problem, since it doesn't affect the release version of Carla.

**Comment by sjoerdvankreel** [comment]
Closing this for now. I'll put a todo item in the list to check out carla 2.6 when it's released. Thanks for re-testing.

**Comment by sjoerdvankreel** [comment]
@SpotlightKid forgot to ask - did you report to falkTX? No idea if the problem is on firefly side or carla, but I suspect this issue will pop up again on the next carla release.

**Comment by SpotlightKid** [comment]
Not, yet, but I will.
